scripts/test_updates.sh

```diff
@@ -5,75 +5,134 @@ set -o pipefail
+docker_pgscript timescaledb-orig /src/test/sql/updates/setup.sql
 docker rm -f timescaledb-orig
 
-docker run -d --name timescaledb-updated -v ${BASE_DIR}:/src -v ${UPDATE_VOLUME}:/var/lib/postgresql/data ${UPDATE_TO_IMAGE}:${UPDATE_TO_TAG}
+docker_run_vol timescaledb-updated ${UPDATE_VOLUME}:/var/lib/postgresql/data ${UPDATE_TO_IMAGE}:${UPDATE_TO_TAG}
 
 wait_for_pg timescaledb-updated
 
 echo "Executing ALTER EXTENSION timescaledb UPDATE"
-docker exec -it timescaledb-updated /bin/bash -c "psql -h localhost -U postgres -d single -c \"ALTER EXTENSION timescaledb UPDATE\""
+docker_pgcmd timescaledb-updated "ALTER EXTENSION timescaledb UPDATE"
 
-wait_for_pg timescaledb-clean
-
-echo "Executing setup script on new version"
-docker exec -it timescaledb-clean /bin/bash -c "psql -h localhost -U postgres -f /src/test/sql/updates/setup.sql"
+docker_exec timescaledb-updated "pg_dump -h localhost -U postgres -Fc single > /tmp/single.sql"
+docker cp timescaledb-updated:/tmp/single.sql ${PGTEST_TMPDIR}/single.sql
 
-echo "Restarting clean container"
-#below is needed so the clean container looks like updated, which has been restarted after the setup script
-#(especially needed for sequences which might otherwise be in different states -- e.g. some backends may have reserved batches)
-docker rm -f timescaledb-clean
-docker run -d --name timescaledb-clean -v ${BASE_DIR}:/src -v ${CLEAN_VOLUME}:/var/lib/postgresql/data ${UPDATE_TO_IMAGE}:${UPDATE_TO_TAG}
 wait_for_pg timescaledb-clean
 
-#same restart of sequences needed on update container in case the update script modified sequences
-docker rm -f timescaledb-updated
-docker run -d --name timescaledb-updated -v ${BASE_DIR}:/src -v ${UPDATE_VOLUME}:/var/lib/postgresql/data ${UPDATE_TO_IMAGE}:${UPDATE_TO_TAG}
-wait_for_pg timescaledb-updated
+echo "Restoring database on new version"
+docker cp ${PGTEST_TMPDIR}/single.sql timescaledb-clean:/tmp/single.sql
+docker_exec timescaledb-clean "createdb -h localhost -U postgres single"
+docker_pgcmd timescaledb-clean "ALTER DATABASE single SET timescaledb.restoring='on'"
+docker_exec timescaledb-clean "pg_restore -h localhost -U postgres -d single /tmp/single.sql"
+docker_pgcmd timescaledb-clean "SELECT restore_timescaledb(); ALTER DATABASE single SET timescaledb.restoring='off'"
 
 echo "Testing"
-docker exec timescaledb-updated /bin/bash \
-  -c "psql -X -v ECHO=ALL -h localhost -U postgres -d single -f /src/test/sql/updates/test-0.1.1.sql" > ${PGTEST_TMPDIR}/updated.out
-docker exec timescaledb-clean /bin/bash \
-  -c "psql -X -v ECHO=ALL -h localhost -U postgres -d single -f /src/test/sql/updates/test-0.1.1.sql" > ${PGTEST_TMPDIR}/clean.out
-
-docker rm -vf timescaledb-updated timescaledb-clean
-
-diff ${PGTEST_TMPDIR}/clean.out ${PGTEST_TMPDIR}/updated.out | tee ${PGTEST_TMPDIR}/update_test.output
+docker_pgdiff timescaledb-updated timescaledb-clean /src/test/sql/updates/test-0.1.1.sql
```

Comment:
Why is this file named with an old version number ? Should it be renamed ?

Reply:
You mean `test-0.1.1.sql`? I actually don't know why this file has this weird name as I didn't create it. My guess is that it is reflecting testing a database that has been updated from 0.1.1...

In any case, if we want to rename (which I think make sense), it should probably be a separate PR.

Reply:
Yeah it probably should be renamed, we didn't even have a version 0.1.1 so that's strange...

Reply:
(A separate PR is fine)